components/indicators/IndicatorListFiltered.tsx

```diff
@@ -692,19 +704,41 @@ const IndicatorListFiltered = (props) => {
                           </IndicatorLink>
                         </IndentableTableCell>
                       )}
-                      {someIndicatorsHaveCategories && (
-                        <IndentableTableCell>
-                          {item.categories.map((cat) => {
-                            if (cat && (shouldDisplayCategory?.(cat) ?? true))
-                              return (
-                                <StyledBadge key={cat.id}>
-                                  {cat.name}
-                                </StyledBadge>
-                              );
-                            return false;
-                          })}
-                        </IndentableTableCell>
-                      )}
+                      {includePlanRelatedIndicators
+                        ? commonCategories.map((commonCategory) => (
+                            <IndentableTableCell
+                              key={`cat-${commonCategory.typeIdentifier}`}
+                            >
+                              {item.categories
+                                .filter(
+                                  (cat) =>
+                                    cat.common &&
+                                    cat.common.type.identifier ===
+                                      commonCategory.typeIdentifier
+                                )
+                                .map((cat) => (
+                                  <StyledBadge key={cat.common.id}>
+                                    {cat.common.name}
+                                  </StyledBadge>
+                                ))}
+                            </IndentableTableCell>
+                          ))
+                        : someIndicatorsHaveCategories && (
```

Comment:
I find the ternary operator included within React markup makes the whole thing a bit hard to read. I wonder if we could either extract the two different variations into separate functions (components) or at the minimum initialize them into a const variable before this code.

I mean instead of:
`condition ? <blah blah blah blah blahz> : <another blah blah blah>`
we would have

```
const ComponentA  = (props) => <blah blah blah>
const ComponentB = (props) => <another blah blah>

...

condition ? <ComponentA props.../> : <ComponentB props.../>
```
for example

Reply:
Refactored the ternary operator within the markup.
Also added some more typing.